src/test/test-parse-util.c

```diff
@@ -922,4 +923,51 @@ TEST(parse_loadavg_fixed_point) {
         assert_se(parse_loadavg_fixed_point("", &fp) == -EINVAL);
 }
 
+TEST(pcr_index_from_string) {
+        assert_se(pcr_index_from_string("core-firmware") == 0);
+        assert_se(pcr_index_from_string("0") == 0);
+        assert_se(pcr_index_from_string("platform-config") == 1);
+        assert_se(pcr_index_from_string("1") == 1);
+        assert_se(pcr_index_from_string("external-code") == 2);
+        assert_se(pcr_index_from_string("2") == 2);
+        assert_se(pcr_index_from_string("external-firmware") == 3);
+        assert_se(pcr_index_from_string("3") == 3);
+        assert_se(pcr_index_from_string("boot-loader") == 4);
+        assert_se(pcr_index_from_string("4") == 4);
+        assert_se(pcr_index_from_string("boot-loader-config") == 5);
+        assert_se(pcr_index_from_string("5") == 5);
+        assert_se(pcr_index_from_string("power-events") == 6);
+        assert_se(pcr_index_from_string("6") == 6);
+        assert_se(pcr_index_from_string("secure-boot-policy") == 7);
+        assert_se(pcr_index_from_string("7") == 7);
+        assert_se(pcr_index_from_string("kernel-initrd") == 9);
+        assert_se(pcr_index_from_string("9") == 9);
+        assert_se(pcr_index_from_string("ima") == 10);
+        assert_se(pcr_index_from_string("10") == 10);
+        assert_se(pcr_index_from_string("kernel-boot") == 11);
+        assert_se(pcr_index_from_string("11") == 11);
+        assert_se(pcr_index_from_string("kernel-config") == 12);
+        assert_se(pcr_index_from_string("12") == 12);
+        assert_se(pcr_index_from_string("systexts") == 13);
+        assert_se(pcr_index_from_string("13") == 13);
+        assert_se(pcr_index_from_string("shim-policy") == 14);
+        assert_se(pcr_index_from_string("14") == 14);
+        assert_se(pcr_index_from_string("system-identity") == 15);
+        assert_se(pcr_index_from_string("15") == 15);
+        assert_se(pcr_index_from_string("hello") == -EINVAL);
+        assert_se(pcr_index_from_string("8") == 8);
+        assert_se(pcr_index_from_string("44") == -EINVAL);
+}
+
+TEST(tpm2_pcr_mask_from_string) {
+uint32_t k = 1024;
+
+assert_se(!tpm2_pcr_mask_from_string("7+power-events", &k));
+assert_se(!tpm2_pcr_mask_from_string("8+boot-loader,11", &k));
```

Comment:
Well, the test test should actually test that the return value make sense.

Reply:
Also, the intiailization of `k` is not needed.

Reply:
> Well, the test test should actually test that the return value make sense.

The function returns 0 on success or -EINVAL if it fails sir

Reply:
> Also, the intiailization of `k` is not needed.

Alright sir.

Reply:
@medhefgo, @keszybz has agreed to "host-platform".
Kindly check his explanation on the fact that TCG PC Client Platform Firmware Profile Specification revision 1.05 says "Previously defined measurements for PCR[6] from the PC Client Implementation Specification for Conventional
BIOS have been deprecated"

We'll use "host-platform".